src/Routing/Route/DashedRoute.php

```diff
@@ -56,11 +56,12 @@ protected function _camelizePlugin($plugin)
      * camelBacked form.
      *
      * @param string $url The URL to parse
+     * @param string $method The HTTP method.
      * @return array|false An array of request parameters, or false on failure.
      */
-    public function parse($url)
+    public function parse($url, $method = '')
```

Comment:
It will :(

Should we add another method and deprecate this one? Another option is using `func_get_args()`